### ISSUE TITLE
Youtube doesn't fetch all subscriptions, ~140 only

### ISSUE THREAD

**Issue by nightshade427** [futo-org/grayjay-android#1310]
### What happened?

I have 500 subscriptions, when I refresh the subscription tab it says getting important 140 subscriptions, and i don't see all the videos for all 500 in subscriptions tab, so I'm always missing a bunch of videos making it hard to use the app since I have to keep going back to YouTube app and compare subscription feed and watch missing videos in YouTube app.

### Grayjay Version

262

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

201

### When do you experience the issue?

- [X] While logged in
- [ ] While logged out
- [ ] N/A

**Comment by Vacknick** [comment]
Same here

**Comment by Kelvin-FUTO** [comment]
This should be an issue that solves itself as you use the app. Especially if you enable background updating.
Grayjay has an internal cache that it builds up for all channels you are subscribed to.
Some channels simply dont upload daily, so making requests would be redundant.
To avoid triggering captchas, we have to limit the amount of requests made by Grayjay.

With 500 subscriptions, I highly recommend you enable background updating so that most of your subscriptions stay relatively up2date.

**Comment by Kelvin-FUTO** [comment]
Alternatively, we have an experimental "Peek Channel Contents" toggle in settings.
This allows Grayjay to mix in recent results for more channels without affecting rate-limiting too much.
Sadly the results from peek channel contents are inferior, and may contain Shorts and such. Hence its not enabled by default, since most people dont have 200+ subscriptions.

**Comment by nightshade427** [comment]
I have used grayjay for about week. I'm seeing missing videos a lot if I compare subscription feed from YouTube app vs grayjay. I found if I go to YouTube feed, look at missing video and note its creator name, then find that creator in creator tab in grayjay, click on creator, then refresh subscription, then the missing video shows up. But, I've had to do that for each missing video in grayjay.

**Comment by TheColorman** [comment]
This is a bit out there, but what about a crowdsourcing feature where updating your subscriptions sends the video info to GrayJay as a sort of global cache, allowing individual clients to decide whether to use the remote cache for each creator, depending on how old it is?
I know this kind of goes against the whole "decentralisation" ideology and can be seen as collecting user data, but I, and I'm sure many others, am fine with contributing usage data for open source projects.

**Comment by Vacknick** [comment]
Having the option to fetch feeds using RSS, similar to how FreeTube does would be nice

**Comment by Kelvin-FUTO** [comment]
> "Peek Channel Contents"

This is basically what the previously mentioned "Peek Channel Contents" already does.
The problem is is that the data is inferior, and may include shorts which is undesireable for most. Hence its disabled by default. To avoid complaints about that.

**Comment by Tayvick** [comment]
Oh, thank you. My bad, sorry

**Comment by nightshade427** [comment]
was the original issue of supporting 500 channels fixed? Just curious as the issue was closed.

**Comment by Zvonimir-FUTO** [comment]
> was the original issue of supporting 500 channels fixed? Just curious as the issue was closed.

Please read Kelvin's comments so I don't regurgitate everything again.

**Comment by nightshade427** [comment]
I have tried with backgrounding and peek and still missing lots of videos when I have 500 channels. 

I have to go to the creator tab, search for the creator of the missing video, view their feed, then go back to the subscription page and refresh for the video to show up. I have to do this hourly to make.the missing videos show up. It makes the app not usable. 

If 500 channels isn't possible to support maybe put in the docs or similar the max limit?

**Comment by Zvonimir-FUTO** [comment]
Thank you for providing feedback. We will look into this again after holidays.

**Comment by JustJinxed** [comment]
> > was the original issue of supporting 500 channels fixed? Just curious as the issue was closed.
> 
> Please read Kelvin's comments so I don't regurgitate everything again but essentially to quote him, "This should be an issue that solves itself as you use the app", "Alternatively, we have an experimental "Peek Channel Contents" toggle in settings" which you can try as well.

This isn't a resolution to the problem. "should" solve itself isn't fixing my subscriptions pane after a week of use. I can't sit there and watch 3 day old videos, and miss out on over 20 different uploads that happened THAT day, that I can see on YouTube's subscription page. Is GrayJay pulling from the subscription page? And if not, isn't there a way to limit how far back it tries to pull? I'm very confused how this is a fix, when I haven't seen anything remedied yet, even though I keep trying.

**Comment by Zvonimir-FUTO** [comment]
> > > was the original issue of supporting 500 channels fixed? Just curious as the issue was closed.
> > 
> > 
> > Please read Kelvin's comments so I don't regurgitate everything again but essentially to quote him, "This should be an issue that solves itself as you use the app", "Alternatively, we have an experimental "Peek Channel Contents" toggle in settings" which you can try as well.
> 
> This isn't a resolution to the problem. "should" solve itself isn't fixing my subscriptions pane after a week of use. I can't sit there and watch 3 day old videos, and miss out on over 20 different uploads that happened THAT day, that I can see on YouTube's subscription page. Is GrayJay pulling from the subscription page? And if not, isn't there a way to limit how far back it tries to pull? I'm very confused how this is a fix, when I haven't seen anything remedied yet, even though I keep trying.

Highlighting few messages from the thread:

>Having the option to fetch feeds using RSS similar to how FreeTube does would be nice

>This is basically what the previously mentioned "Peek Channel Contents" already does.
>The problem is is that the data is inferior, and may include shorts which is undesireable for most. Hence its disabled by default. To avoid complaints about that.

This is supposed to fix the problem but @nightshade427 says it still doesn't work hence we reopened the issue to look further into it.

And as Kelvin also mentioned:

>To avoid triggering captchas, we have to limit the amount of requests made by Grayjay.

**Comment by ZacWarham** [comment]
Happening to me also. Not sure how to check how many subscriptions I have but I think it was around 120 from importing. 

I enabled the peek channel contents which retrieved one missing video but I'm concerned about not being able to trust it now and that it still may be missing others. 

Also, enabling this gave me shorts as expected, that doesn't bother me as long as everything is showed but it appears not all shorts are loaded and it's loading them with the wrong datetime stamp. It said a short was 4 minutes old but it was actually uploaded a few days ago now

**Comment by VoxelPrismatic** [comment]
> I have tried with backgrounding and peek feature and still missing lots of videos when I have 500 channels. 
> 
> I have to go to the creator tab, search for the creator of the missing video, view their feed, then go back to the subscription page and refresh for the video to show up. I have to do this hourly to make the missing videos show up (there are 10-20 missing videos each time). It makes the app not usable. 
> 
> If 500 channels isn't possible to support maybe put in the docs or similar the max limit?

In the general app settings, you can tell it to refresh more often.

It's not so much of a "max limit," but rather YouTube prevents you from requesting too many at once. Grayjay does a good job of caching so you don't notice.

![ResizedImage_2025-01-31_20-21-23_1269.png](https://github.com/user-attachments/assets/2c40a982-779e-4e7b-9503-b71a98117838)

**Comment by ykuksenko** [comment]
As a possible mitigation for missed videos. Can grayjay show videos in order of discovery rather than youtube publication date?
For example if a video was published 3 days ago but grayjay discovers it today can it be placed at the top of the subscription list rather than pages and pages down among 3 day old videos where I will likely never see it?

**Comment by JustJinxed** [comment]
@Zvonimir-FUTO Have they considered maybe using RSS for people who have 50+ subscriptions, or making it an option? I know other open clients (like freetube) have had to give up on requesting direct data for this very reason.

**Comment by Zvonimir-FUTO** [comment]
Could you guys update to v297 and test again?

**Comment by defiling9046** [comment]
> Could you guys update to v297 and test again?

has alot changed since then?

also has nobody mentioned the "use subscription exchange" feature in Grayjays settings as I feel like this problem will solve itself out depending on how many people use it

**Comment by Zvonimir-FUTO** [comment]
@defiling9046 yeah, there are now Android streams which hopefully should work. 

Subscription exchange doesn't exchange the video data (whole video), it exchanges video metadata (links, name, etc.)

**Comment by defiling9046** [comment]
> [@defiling9046](https://github.com/defiling9046) yeah, there are now Android streams which hopefully should work.
> 
> Subscription exchange doesn't exchange the video data (whole video), it exchanges video metadata (links, name, etc.)

Feel free to correct me as i'm prob misunderstanding as I don't even know how to code lol but I thought the subscription exchange would maybe transfer my subscriptions (the video,the author of the video and thumbnail) to a centralised server and if someone else had the same subscription as me then instead of going over the 140 threshold and getting a captcha the grayjay app would look into the centralised server to see if there are any other subscriptions that need updating.


"Uses a crowd-sourced server to significantly reduce the required requests for subscriptions, in exchange you submit your subscriptions to the server."

**Comment by Zvonimir-FUTO** [comment]
You are right for most of it except sending the whole video. The stream is still being pulled from platform servers.

**Comment by name-tar-xz** [comment]
> To avoid triggering captchas, we have to limit the amount of requests made by Grayjay.

Most other 3rd party youtube clients (NewPipe and its forks, LibreTube) can easily fetch a feed of 400+ subscriptions without triggering any captchas/ratelimits. 
Does grayjay do something different or is this completely arbitrary? If so please allow the user to set their upper limit for fetched subscribers and default it to 140 to retain backward compatibility.

**Comment by Zvonimir-FUTO** [comment]
> > To avoid triggering captchas, we have to limit the amount of requests made by Grayjay.
> 
> Most other 3rd party youtube clients (NewPipe and its forks, LibreTube) can easily fetch a feed of 400+ subscriptions without triggering any captchas/ratelimits. Does grayjay do something different or is this completely arbitrary? If so please allow the user to set their upper limit for fetched subscribers and default it to 140 to retain backward compatibility.

This is because Grayjay handles this differently since it's platform agnostic so when we fetch your subscriptions, we don't fetch  `https://www.youtube.com/feed/subscriptions`, we fetch channels pages you are subscribed to.

Keep in mind that Grayjay user can use plugins like Youtube without using an account and subscribe to channels. Or use an account but never subscribing with their Youtube account.

**Comment by Zvonimir-FUTO** [comment]
~~We headed a bit off topic and it seems there is no issue with this anymore. Closing the issue.~~

**Comment by name-tar-xz** [comment]
> This is because Grayjay handles this differently since it's platform agnostic so when we fetch your subscriptions, we don't fetch https://www.youtube.com/feed/subscriptions, we fetch channels pages you are subscribed to.

All of the alternatives I listed fetch subscriptions in the same way as grayjay. Without any logging in and by fetching each channels page individually.
 
> We headed a bit off topic and it seems there is no issue with this anymore. Closing the issue.

The issue clearly still exsists. Not sure what about any of thay was off topic either.

**Comment by Zvonimir-FUTO** [comment]
Oh my apologies, I misread the issue thinking it's about downloading videos downloads them in 140p only...
I guess that happens after hours and hours of going through issues.

You are right, you weren't off topic, I was haha

**Comment by ThunderHen** [comment]
Just a thought I'm not sure if it would help or atleast something similar but I know over time i have had channels that have stopped uploading what if there was a way to go through subscribed channels and remove channels that have not uploaded anything in a while this could mabe help reduce the number of YouTube channels Grayjay trys to fech overtime. Or is Grayjay feching channel data in a way where this would not be helpful?

**Comment by skinnyhastrup-design** [comment]
I always thought this was some way of add revenue to ad-driven channels, that the "freeloaders" don't get a notification through non-YT even though they subscribe on the official YT account.
I don't get notification from all the big channels on grayjay, like Kitboga, Scammer Payback or Rick Beato, but most of the smaller ones still notify me.
